lib/Test/APIcast.pm

```diff
@@ -38,21 +38,38 @@ our @EXPORT = qw( get_random_port );
 
 our @PORTS = ();
 
+my $prove_filename_lock="/tmp/prove_lock";
+
+sub wait_file_unlock{
+	while (-e "$prove_filename_lock") {
+        sleep(rand(1))
+	}
+}
+
+sub file_lock {
+	`touch $prove_filename_lock`;
+}
+
+sub file_unlock {
+	unlink "$prove_filename_lock";
+}
+
 sub get_random_port {
     my $tries = 1000;
     my $ServerPort;
-
+    wait_file_unlock();
+    file_lock();
     for (my $i = 0; $i < $tries; $i++) {
         my $port = int(rand 60000) + 1025;
 
         my $sock = IO::Socket::INET->new(
-            LocalAddr => $Test::Nginx::Util::ServerAddr,
```

Comment:
I think this should be there as that's where nginx will try to listen.

Reply:
```

localhost.localdomain ➜  APIcast git:(Issue-1010) ✗ cat t/servroot_10103__898/conf/nginx.conf  | grep list
        listen          10103;
localhost.localdomain ➜  APIcast git:(Issue-1010) ✗
```

Is listen on all IP, so this is more real IMHO.